src/_P036_FrameOLED.ino

```diff
@@ -600,30 +621,74 @@ boolean Plugin_036(byte function, struct EventStruct *event, String& string)
 
     case PLUGIN_WRITE:
       {
-        // FIXME TD-er: This one is not using parseString* function
-        String tmpString  = string;
-        int argIndex = tmpString.indexOf(',');
-        if (argIndex)
-          tmpString = tmpString.substring(0, argIndex);
-        if (tmpString.equalsIgnoreCase(F("OLEDFRAMEDCMD")) && display)
-        {
-          success = true;
-          argIndex = string.lastIndexOf(',');
-          tmpString = string.substring(argIndex + 1);
-          if (tmpString.equalsIgnoreCase(F("Off")))
-            P36_setContrast(P36_CONTRAST_OFF);
-          else if (tmpString.equalsIgnoreCase(F("On")))
-            display->displayOn();
-          else if (tmpString.equalsIgnoreCase(F("Low")))
-            P36_setContrast(P36_CONTRAST_LOW);
-          else if (tmpString.equalsIgnoreCase(F("Med")))
-            P36_setContrast(P36_CONTRAST_MED);
-          else if (tmpString.equalsIgnoreCase(F("High")))
-            P36_setContrast(P36_CONTRAST_HIGH);
+        String log     = "";
+        String command = parseString(string, 1);
+        String subcommand = parseString(string, 2);
+        int LineNo = event->Par1;
+
+        if ((command == F("oledframedcmd")) && display) {
+          if (subcommand == F("display"))
+          {
+            // display functions
+            String para1 = parseString(string, 3);
+            if (para1 == F("on")) {
+              success = true;
+              displayTimer = PCONFIG(4);
+              display->displayOn();
+              UserVar[event->BaseVarIndex] = 1;      //  Save the fact that the display is now ON
+            }
+            if (para1 == F("off")) {
+              success = true;
+              displayTimer = 0;
+              display->displayOff();
+              UserVar[event->BaseVarIndex] = 0;      //  Save the fact that the display is now OFF
+            }
+            if (para1 == F("low")) {
+              success = true;
+              P36_setContrast(P36_CONTRAST_LOW);
+            }
+            if (para1 == F("med")) {
+              success = true;
+              P36_setContrast(P36_CONTRAST_MED);
+            }
+            if (para1 == F("high")) {
+              success = true;
+              P36_setContrast(P36_CONTRAST_HIGH);
+            }
+            // log += String(F("[P36] Display: ")) + String(para1) + String(F(" Success:")) + String(success);
+            // addLog(LOG_LEVEL_INFO, log);
         }
-        break;
-      }
+        else if ((LineNo >= 1) && (LineNo <= (P36_Nlines+1)))
+          {
+            // content functions
+            success = true;
+            String NewContent = parseStringKeepCase(string, 3);
+            P036_displayLines[LineNo-1] = NewContent;
```

Comment:
You allow LineNo == P36_Nlines+1

So `P036_displayLines[LineNo-1]` can become `P036_displayLines[P36_Nlines]`, which is out of bounds

Suggestion:
`else if ((LineNo > 0) && (LineNo <= P36_Nlines))`

Reply:
Done!